src/stp/new_plays/GetBallRisky.cpp

```diff
@@ -0,0 +1,110 @@
+
+    keepPlayInvariants.clear();
+    keepPlayInvariants.emplace_back(std::make_unique<invariant::NormalPlayGameStateInvariant>());
+    keepPlayInvariants.emplace_back(std::make_unique<invariant::BallIsFreeInvariant>());
+    keepPlayInvariants.emplace_back(std::make_unique<invariant::WeHaveMajorityInvariant>());
+
+    roles = std::array<std::unique_ptr<Role>, rtt::ai::Constants::ROBOT_COUNT()>{std::make_unique<role::Keeper>(role::Keeper("keeper")),
+                                                                                 std::make_unique<role::BallGetter>(role::BallGetter("ball_getter")),
+                                                                                 std::make_unique<role::PassReceiver>(role::PassReceiver("receiver_0")),
+                                                                                 std::make_unique<role::PassReceiver>(role::PassReceiver("receiver_1")),
+                                                                                 std::make_unique<role::PassReceiver>(role::PassReceiver("receiver_2")),
+                                                                                 std::make_unique<role::Defender>(role::Defender("defender_0")),
+                                                                                 std::make_unique<role::Defender>(role::Defender("defender_1")),
+                                                                                 std::make_unique<role::Defender>(role::Defender("defender_2")),
+                                                                                 std::make_unique<role::Defender>(role::Defender("midfielder_0")),
+                                                                                 std::make_unique<role::Defender>(role::Defender("midfielder_1")),
+                                                                                 std::make_unique<role::Defender>(role::Defender("midfielder_2"))};
+}
+
+uint8_t GetBallRisky::score(world_new::World* world) noexcept { return 120; }
+
+void GetBallRisky::calculateInfoForRoles() noexcept {
+    auto enemyRobots = world->getWorld()->getThem();
+    auto enemyAttacker = world->getWorld()->getRobotClosestToBall(world_new::them);
+
+    enemyRobots.erase(std::remove_if(enemyRobots.begin(), enemyRobots.end(), [&](const auto enemyRobot) -> bool { return enemyRobot->getId() == enemyAttacker->getId(); }));
+
+    auto enemyClosestToGoal = world->getWorld()->getRobotClosestToPoint(field.getOurGoalCenter(), enemyRobots);
+
+    stpInfos["keeper"].setEnemyRobot(world->getWorld()->getRobotClosestToBall(world_new::them));
+
+    // TODO: determine better future receive positions
+    stpInfos["receiver_0"].setPositionToMoveTo(Vector2(field.getFieldLength() / 4, field.getFieldWidth() / 4));
+    stpInfos["receiver_1"].setPositionToMoveTo(Vector2(field.getFieldLength() / 4, -field.getFieldWidth() / 4));
+    stpInfos["receiver_2"].setPositionToMoveTo(Vector2(field.getFieldLength() / 4, 0));
```

Comment:
Can I get input on how to determine good pass positions without pagmo? probably needs to share the algorithm with the actual pass play

Reply:
Yeah, I'm trying to find a good algorithm. I think once the new pass play (without pagmo) is done we will have a bit more information about this